staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -185,46 +185,37 @@ func (r *ScaleREST) Update(ctx genericapirequest.Context, name string, objInfo r
 
 // scaleFromCustomResource returns a scale subresource for a customresource.
 func scaleFromCustomResource(cr *unstructured.Unstructured, specReplicasPath, statusReplicasPath, labelSelectorPath string) (*scalescheme.Scale, error) {
-	var specReplicas, statusReplicas int64
-	var customResourceLabelSelector []byte
-	var ok bool
-
-	replicas := apiextensions.GetJSONPath(cr.UnstructuredContent(), specReplicasPath)
-	if replicas != nil {
-		specReplicas, ok = replicas.(int64)
-		if !ok {
-			return nil, fmt.Errorf("invalid spec replicas value: %v", replicas)
-		}
-		if specReplicas < 0 {
-			return nil, fmt.Errorf("spec replicas value should be a non-negative integer")
-		}
+	specReplicasPath = strings.TrimPrefix(specReplicasPath, ".") // ignore leading period
+	specReplicas, ok := unstructured.NestedInt64(cr.UnstructuredContent(), strings.Split(specReplicasPath, ".")...)
+	if !ok {
+		return nil, fmt.Errorf("invalid spec replicas value: %v", specReplicas)
+	}
+	if specReplicas < 0 {
+		return nil, fmt.Errorf("spec replicas value should be a non-negative integer")
 	}
 
-	replicas = apiextensions.GetJSONPath(cr.UnstructuredContent(), statusReplicasPath)
-	if replicas != nil {
-		statusReplicas, ok = replicas.(int64)
-		if !ok {
-			return nil, fmt.Errorf("invalid status replicas value: %v", replicas)
-		}
-		if statusReplicas < 0 {
-			return nil, fmt.Errorf("status replicas value should be a non-negative integer")
-		}
-	} else {
+	statusReplicasPath = strings.TrimPrefix(statusReplicasPath, ".") // ignore leading period
+	statusReplicas, ok := unstructured.NestedInt64(cr.UnstructuredContent(), strings.Split(statusReplicasPath, ".")...)
+	if !ok {
 		statusReplicas = 0
 	}
+	if statusReplicas < 0 {
+		return nil, fmt.Errorf("status replicas value should be a non-negative integer")
+	}
 
-	ls := apiextensions.GetJSONPath(cr.UnstructuredContent(), labelSelectorPath)
 	scaleLabelSelector := &metav1.LabelSelector{}
-	if ls != nil {
+	var customResourceLabelSelector []byte
+	ls, ok := unstructured.NestedFieldCopy(cr.UnstructuredContent(), strings.Split(labelSelectorPath, ".")...)
```

Comment:
This needs fixing. Will do that.

Reply:
Oops, done in https://github.com/kubernetes/kubernetes/pull/55168/commits/6f36088c3ae8216c04ed599d640c339a48f2cac9. Should be fine after a squash.